miasm2/expression/expression.py

```diff
@@ -318,12 +318,10 @@ def replace_expr(self, dct=None):
         @dct: dictionary of Expr -> *
         """
         if dct is None:
-            dct = {}
+            return self
 
         def my_replace(expr, dct):
-            if expr in dct:
-                return dct[expr]
-            return expr
+            return dst.get(expr, expr)
 
         return self.visit(lambda expr: my_replace(expr, dct))
```

Comment:
Maybe since your (good) proposal to implement `my_replace()` is as simple as a function call, you could remove it and use here:
```Python
        return self.visit(lambda expr: dct.get(expr, expr))
```